src/otaku.cpp

```diff
@@ -129,6 +183,54 @@ struct obj_data *make_otaku_deck(struct char_data *ch) {
   return new_deck;
 }
 
+void update_otaku_deck(struct char_data *ch, struct obj_data *cyberdeck) {
+  int mpcp = GET_OTAKU_MPCP(ch);
+
+  GET_CYBERDECK_MPCP(cyberdeck) = MIN(12, mpcp);
+  GET_CYBERDECK_HARDENING(cyberdeck) = MIN(get_otaku_wil(ch), get_otaku_wil(ch) / 2 + GET_ECHO(ch, ECHO_IMPROVED_HARD));
+  GET_CYBERDECK_ACTIVE_MEMORY(cyberdeck) = 0; // Otaku do not have active memory.
+  GET_CYBERDECK_TOTAL_STORAGE(cyberdeck) = 0;
+  GET_CYBERDECK_RESPONSE_INCREASE(cyberdeck) = MIN(mpcp * 1.5, get_otaku_rea(ch) + GET_ECHO(ch, ECHO_IMPROVED_REA));
+  GET_CYBERDECK_IO_RATING(cyberdeck) = MIN(get_otaku_int(ch) * 200, (get_otaku_int(ch) * 100) + (GET_ECHO(ch, ECHO_IMPROVED_IO) * 100));
```

Comment:
Remember that `MIN` and `MAX` are hacky macros that execute their contents between 1-2 times per branch. This expands to the equivalent of this function:

```c++
if (get_otaku_int(ch) * 200 <= (get_otaku_int(ch) * 100) + (GET_ECHO(ch, ECHO_IMPROVED_IO) * 100)) {
  return get_otaku_int(ch) * 200;
} else {
  return (get_otaku_int(ch) * 100) + (GET_ECHO(ch, ECHO_IMPROVED_IO) * 100);
}
```

These are relatively lightweight functions and are idempotent, but if update_otaku_deck() gets called a lot in a single tick, it'll contribute to the tick lag. Best practice for a single-threaded game like this is to save the output of reused idempotent functions to a local variable where possible.

Reply:
Not a required change, just a point for future dev work.